theme/scripts/logo.ts

```diff
@@ -7,19 +9,37 @@ export default (function() {
   const Logo = {
     SELECTORS: {
       logoText: "[data-logo-text]",
+      logoTextMobile: "[data-logo-text-mobile]",
     },
     CONSTANTS: {
       index: 'index'
     },
+    STATE: {
+      setInterval: null
+    },
 
     init() {
       const logoEl: HTMLElement = document.querySelector(Logo.SELECTORS.logoText);
-      Logo.bindEventListeners(logoEl);
+      const logoElMobile: HTMLElement = document.querySelector(Logo.SELECTORS.logoTextMobile);
+      Logo.bindEventListeners(logoEl, logoElMobile);
     },
 
-    bindEventListeners(logoEl: HTMLElement) {
+    bindEventListeners(logoEl: HTMLElement, logoElMobile: HTMLElement) {
       logoEl.addEventListener('mouseenter', () => Logo.randomizeLogo(logoEl));
-      logoEl.addEventListener('mouseleave', () => Logo.resetLogo(logoEl))
+      logoEl.addEventListener('mouseleave', () => Logo.resetLogo(logoEl));
+      window.addEventListener('DOMContentLoaded', () => Logo.checkBreakpointAndRandomizeLogo(logoElMobile));
+      window.addEventListener('resize', () => Logo.checkBreakpointAndRandomizeLogo(logoElMobile));
+    },
+
+    checkBreakpointAndRandomizeLogo(logoElMobile: HTMLElement) {
+      if (isMobile) {
```

Comment:
We'll want to call `isMobile()`, otherwise this will always evaluate to true since a `func()` is considered truthy in js.

Since we're not stashing the previous `isMobile()` and comparing it, we'll want to also check that an existing setInterval has not already been set, before setting it, we'll also want to check that there is an interval

```
//Mobile
if (isMobile()) {
// If an interval has already been set, return
 if (Logo.STATE.setIntervalId) return;
 // An interval has not been set
  Logo.STATE.setInterval = setInterval(() => {
   Logo.randomizeLogo(logoElMobile)
  }, 2000); 
return;
} else {
// Desktop 
Logo.clearInterval(); 
// We can do this, because clearInterval already checks whether a setIntervalId has been set before attempting to clear it :)
}


```

Reply:
Hahah yeah discovered this when it was always falling into the isMobile case 😏 Thank you for these helpful notes!!